child_compassion/wizard/project_description_de.py

```diff
@@ -0,0 +1,175 @@
+
+        wall_mat = [mat.value_de if mat.value_de else mat.value_en
+                    for mat in project.wall_material_ids]
+
+        roof_mat = [mat.value_de if mat.value_de else mat.value_en
+                    for mat in project.roof_material_ids]
+
+        string = u"Die Häuser sind typischerweise mit "
+
+        if wall_mat[0] and floor_mat[0] and roof_mat[0]:
+            string += (u"%s gebaut und haben %s, sowie %s. " % (
+                       wall_mat[0], floor_mat[0], roof_mat[0]))
+
+        elif ((wall_mat[0] and floor_mat[0]) or (wall_mat[0] and
+              roof_mat[0]) or (floor_mat[0] and roof_mat[0])):
+            if not wall_mat[0]:
+                string += (u"%s gebaut und haben %s. " % (
+                           floor_mat[0], roof_mat[0]))
+            elif not floor_mat[0]:
+                string += (u"%s gebaut und haben %s. " % (
+                           wall_mat[0], roof_mat[0]))
+            elif not roof_mat[0]:
+                string += (u"%s gebaut und haben %s. " % (
+                           wall_mat[0], floor_mat[0]))
+
+        elif (wall_mat[0] or floor_mat[0] or roof_mat[0]):
+            if wall_mat[0]:
+                string += u"%s gebaut. " % wall_mat[0]
+            elif floor_mat[0]:
+                string += u"%s gebaut. " % floor_mat[0]
+            elif roof_mat[0]:
+                string += u"%s gebaut. " % roof_mat[0]
+        else:
+            string = ""
+
+        return string
+
+    @classmethod
+    def _gen_primary_diet_de(cls, cr, uid, project, context=None):
+        """ Generate spoken languages(s) and primary diet, there are
+            no specificities in this part
+        """
+        primary_diet = [diet.value_de if diet.value_de else diet.value_en
+                        for diet in project.primary_diet_ids]
+
+        spoken_languages = [lang.value_de if lang.value_de else lang.value_en
+                            for lang in project.spoken_languages_ids]
+
+        string = (u"Die ethnischen Merheiten %s %s" % (
+                  u"sind" if len(spoken_languages) > 1 else u"ist",
+                  cls._gen_list_string(spoken_languages, ', ', ' und ')))
+
+        if spoken_languages[0]:
+            string += (u" und die meist gesprochene Sprache ist %s. " % (
+                       spoken_languages[0]))
+        else:
+            string += ". "
+
+        string += (u"Die regionale Ernährung beinhaltet %s. " % (
+                   cls._gen_list_string(primary_diet, ', ', ' und ')))
+
+        return string
+
+    @classmethod
+    def _gen_health_prob_de(cls, cr, uid, project, context=None):
+        """ Generate health problemes of this region, there
+            are no specificities in this part
+        """
+        health_prob = [prob.value_de if prob.value_de else prob.value_en
+                       for prob in project.health_problems_ids]
+
+        string = (u"Verbereitete Gesundheitsporbleme %s %s. " % (
```

Comment:
What if health_prob is empty? Is the spec to always have the sentence, or can the sentence be dropped if no information is available ?